Windows builds have stopped working again

### Describe the bug

As per the title. I can't figure out what the exact cause is, the only PR in the most recent release (https://github.com/cataclysmbnteam/Cataclysm-BN/releases/tag/2024-02-07) that had any potentially-relevant failures was https://github.com/cataclysmbnteam/Cataclysm-BN/pull/3930 and the error list SEEMED to be the same benign unit conversion issues we've already had in npcmove.cpp for a while already. 

Build log for the MVSC task thingy:
```
Run vcpkg integrate install
Applied user-wide integration for this vcpkg root.
CMake projects should use: "-DCMAKE_TOOLCHAIN_FILE=D:/a/Cataclysm-BN/b/vcpkg/scripts/buildsystems/vcpkg.cmake"

All MSBuild C++ projects can now #include any installed libraries. Linking will be handled automatically. Installing new libraries will make them instantly available.
vcpkg has crashed. Please create an issue at https://github.com/microsoft/vcpkg containing a brief summary of what you were trying to do and the following information.

Version=2023-06-08-19a3cb921a585fce63c0651f56a991a5a08bdc60
EXCEPTION=bad allocation
CMD=
D:\a\Cataclysm-BN\b\vcpkg\vcpkg.exe|
integrate|
install|
Error: Process completed with exit code 1.
```

### Steps To Reproduce

1. Exist in this cursed timeline.
2. Check the daily release.

### Screenshots

_No response_

### Versions and configuration

Best I got, building from current main branch's code reports the above-mentioned unit loss of precision complaints but compiles just fine:
```
- OS: Windows
    - OS Version: 10.0.19041.450 (2004)
- Game Version: 8c6a259796fb [64-bit]
- Graphics Version: Tiles
- LAPI Version: 2
- Game Language:  []
- Mods loaded: [
    Bright Nights [bn],
    No Fungal Monsters [No_Fungi],
    Disable NPC Needs [no_npc_food],
    Manual Bionic Installation [manualbionicinstall],
    Stats Through Kills [stats_through_kills]
]
```

### Additional context

The errors added by the aforementioned PR:
[message.txt](https://github.com/cataclysmbnteam/Cataclysm-BN/files/14203642/message.txt)

--- a/.github/workflows/msvc-full-features-cmake.yml
+++ b/.github/workflows/msvc-full-features-cmake.yml
@@ -16,6 +16,21 @@ on:
     - '!tools/format/**'
     - 'utilities/**'
     - 'scripts/**'
+  pull_request:
+    branches:
+    - main
+    paths-ignore:
+    - 'android/**'
+    - 'build-data/osx/**'
+    - 'doc/**'
+    - 'doxygen_doc/**'
+    - 'gfx/**'
+    - 'lang/**'
+    - 'lgtm/**'
+    - 'tools/**'
+    - '!tools/format/**'
+    - 'utilities/**'
+    - 'scripts/**'
 
 # We only care about the latest revision, so cancel previous instances.
 concurrency:

